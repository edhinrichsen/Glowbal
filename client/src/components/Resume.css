
.resume {
  text-align: center;
  height: calc(100vh - 116px);
  width: 100%;
  max-width: 600px;
  position: -webkit-sticky; /* Safari */
  position: sticky;
  top: var(--unit-margin);

}
.resume-badges {
  display: inline-grid;
  grid-template-columns: auto auto;
}
.resume-badges ul {
  padding: 0;
  list-style-type: none;
}
.resume-badges li {
  cursor: pointer;
  padding: auto;
  text-align: center;
  margin: 5px;
  background-color: var(--background-color-secondary);
  box-shadow: var(--drop-shadow);
  border-radius: 100px;
  /* border: 1px solid; */
  width: 30px;
  /* height: 30px; */
  line-height: 30px;
  display: inline-block;
}

.resume-hero {
  background-image: url(../placeholder-image.png);
  background-size: cover;
  background-position: bottom;
  width: 100%;
  height: 160px;

  /* transition: box-shadow 0.5s, border-color 0.5s, width 1s, height 0.5s 0s, */
    /* margin 0.5s, opacity 0.2s; */
}

.resume-picture {
  margin: auto;
  margin-top: 0;
  margin-bottom: 0;
  background: var(--gradient);
  background-position: center;
  background-size: cover;

  width: 96px;
  height: 96px;
  border-radius: 96px;
  padding: 16px;
  /* position: relative; */
  margin-top: -50px;
 
}
.resume-name {
  padding-top: 8px;
  /* position: relative; */
  /* top: -50px; */
  margin: 0;
}

.resume-profile {
  text-align: left;
  /* padding: 16px;  */
}
