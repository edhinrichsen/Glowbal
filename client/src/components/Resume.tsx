import React, { useState } from 'react';

import { Select, Space, Input, Tooltip } from 'antd';
import { CloseOutlined, MenuOutlined, UserOutlined } from '@ant-design/icons';

import Emoji from './Emoji';

import './Resume.css';
import ResumeEntry from './ResumeEntry';

const { Option } = Select;

type ResumeProps = {
    name: string,
    profile: string,
    student: string,
    location: string,
    work: boolean,
}

function Resume(props: ResumeProps) {

    const studentbadges = () => {

        if (props.student !== "") {
            return (
                <Tooltip title={props.student} placement="bottom"><li><Emoji symbol="🧑‍🎓" label="Student"/></li></Tooltip>
            )
        } else {
            return;
        }

        
    }
    
    const workbadges = () => {

        if (props.work) {
            return (
                <Tooltip title="Open to work opportunities" placement="bottom"><li><Emoji symbol="✅" label="Open to work opportunities"/></li></Tooltip>
            )
        } else {
            return;
        }
        
    }
    
    const locationbadges = () => {

        if (props.location !== "") {
            return (
                <Tooltip title={props.location} placement="bottom"><li><Emoji symbol="🌏" label="Location"/></li></Tooltip>
            )
        } else {
            return;
        }
        
    }

    const [isOpen, setIsOpen] = useState(true);

    if (!isOpen) {
        return (
            <Tooltip title="Browse Options" placement="right" trigger={window.innerWidth <= 800 ? 'click' : 'hover'}>
                <div className="floatingHamburger" onClick={() => setIsOpen(!isOpen)}>
                    <MenuOutlined />
                </div>
            </Tooltip>
        )
    }

    return (
        <div className="container-primary floating-box resume">
            
                {/* <div style={{ display: 'flex', justifyContent: 'space-between' }}> */}
                {/* <Tooltip title="Hide" placement="bottom">
                    <div className="exit-button" onClick={() => setIsOpen(!isOpen)}></div>
                </Tooltip> */}

                    <div className="container-secondary resume-hero"></div>
                    <div className="container-secondary resume-picture">
                    <UserOutlined style={{ fontSize: '64px', color: '#fff' }} />
                    </div>
                    <div className="resume-name">
                    <h1><strong>{props.name}</strong></h1>

                    <div className="resume-badges">
                        <ul>
                            {studentbadges()}
                            {workbadges()}
                            {locationbadges()}
                        </ul>
                    </div>
                    
                    </div>
                    <div className="resume-profile">
                        {props.profile} 
                    </div>
                    <ResumeEntry title="Skills" display="inline" 
                        entries={[
                            {date: "", text: "HTML", subText: ""},
                            {date: "", text: "CSS", subText: ""},
                            {date: "", text: "JavaScript", subText: ""},
                            {date: "", text: "React.js", subText: ""},
                            {date: "", text: "Python", subText: ""},
                            {date: "", text: "C", subText: ""},
                            {date: "", text: "Java", subText: ""},
                            {date: "", text: "Graphic Design", subText: ""},
                            {date: "", text: "Photoshop", subText: ""},
                            {date: "", text: "Illustrator", subText: ""}
                        ]}
                    />
                    <ResumeEntry title="Experience" display="block"
                        entries={[
                            {date: "2020", text: "Internship at a Company", subText: "as a Full Stack Developer"},
                            {date: "2019", text: "Internship at a different Company", subText: "as a Front end Developer"},
                            {date: "2018-Current", text: "Part time job", subText: ""}
                        ]}
                    />

                    <ResumeEntry title="Achievements" display="inline"
                        entries={[
                            {date: "2020", text: "First Cass Honours", subText: ""},
                            {date: "2020", text: "First in Hackathon", subText: ""}
                        ]}
                    />

                    <ResumeEntry title="Education" display="block"
                        entries={[
                            {date: "2018-20", text: "The University of Melbourne", subText: "Bachelor of Science - Major in computing and software systems"},
                            {date: "2017", text: "School College", subText: "Completed International Baccalaureate Diploma"}
                        ]}
                    />

                    
                    
            
                    
                    

                {/* </div> */}

                

              

                

        </div>
    )
}

export default Resume;