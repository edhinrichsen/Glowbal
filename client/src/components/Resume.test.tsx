import React from 'react';
import { shallow } from 'enzyme';
import Resume from './Resume';

describe('Testing Resume', () => {

  it('renders without crashing', () => {
    const component = shallow(
      <Resume
        name="Jane Doe"
        profile="I am a capable and creative computer science student with a flair for problem solving. I have strong technical, interpersonal and communication skills and am aiming to pursue a career in software engineering & design. "
        student="The University of Melbourne"
        location="Melbourne, Australia"
        work={true}
      />);
    expect(component).toMatchSnapshot();
    component.unmount();
  });

})
