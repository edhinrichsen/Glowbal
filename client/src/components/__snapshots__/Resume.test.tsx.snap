// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Resume renders without crashing 1`] = `
<div
  className="container-primary resume container-scroll"
>
  <div
    className="container-secondary resume-hero"
  />
  <div
    className="container-secondary resume-picture"
  >
    <ForwardRef(UserOutlined)
      style={
        Object {
          "color": "#fff",
          "fontSize": "64px",
        }
      }
    />
  </div>
  <div
    className="resume-name"
  >
    <h1>
      <strong>
        <TextInput
          editable={true}
          onChange={[Function]}
          text="Jane Doe"
        />
      </strong>
    </h1>
    <div
      className="resume-badges"
    >
      <ul>
        <Tooltip
          arrowPointAtCenter={false}
          autoAdjustOverflow={true}
          mouseEnterDelay={0.1}
          mouseLeaveDelay={0.1}
          placement="bottom"
          title="The University of Melbourne"
          transitionName="zoom-big-fast"
        >
          <li>
            <Emoji
              label="Student"
              symbol="🧑‍🎓"
            />
          </li>
        </Tooltip>
        <Tooltip
          arrowPointAtCenter={false}
          autoAdjustOverflow={true}
          mouseEnterDelay={0.1}
          mouseLeaveDelay={0.1}
          placement="bottom"
          title="Open to work opportunities"
          transitionName="zoom-big-fast"
        >
          <li>
            <Emoji
              label="Open to work opportunities"
              symbol="✅"
            />
          </li>
        </Tooltip>
        <Tooltip
          arrowPointAtCenter={false}
          autoAdjustOverflow={true}
          mouseEnterDelay={0.1}
          mouseLeaveDelay={0.1}
          placement="bottom"
          title="Melbourne, Australia"
          transitionName="zoom-big-fast"
        >
          <li>
            <Emoji
              label="Location"
              symbol="🌏"
            />
          </li>
        </Tooltip>
      </ul>
    </div>
  </div>
  <div
    className="resume-profile"
  >
    <TextInput
      editable={true}
      multiline={true}
      onChange={[Function]}
      padding="10px"
      text="I am a capable and creative computer science student with a flair for problem solving. I have strong technical, interpersonal and communication skills and am aiming to pursue a career in software engineering & design. "
    />
  </div>
  <ResumeEntry
    display="inline"
    entries={
      Array [
        Object {
          "text": "HTML",
        },
        Object {
          "text": "CSS",
        },
        Object {
          "text": "JavaScript",
        },
        Object {
          "text": "React.js",
        },
        Object {
          "text": "Python",
        },
        Object {
          "text": "C",
        },
        Object {
          "text": "Java",
        },
        Object {
          "text": "Graphic Design",
        },
        Object {
          "text": "Photoshop",
        },
        Object {
          "text": "Illustrator",
        },
      ]
    }
    isMyProfile={true}
    title="Skills"
  />
  <ResumeEntry
    display="block"
    entries={
      Array [
        Object {
          "date": "2020",
          "subText": "as a Full Stack Developer",
          "text": "Internship at a Company",
        },
        Object {
          "date": "2019",
          "subText": "as a Front end Developer",
          "text": "Internship at a different Company",
        },
        Object {
          "date": "2018-Current",
          "text": "Part time job",
        },
      ]
    }
    isMyProfile={true}
    title="Experience"
  />
  <ResumeEntry
    display="inline"
    entries={
      Array [
        Object {
          "date": "2020",
          "text": "First Cass Honours",
        },
        Object {
          "date": "2020",
          "text": "First in Hackathon",
        },
      ]
    }
    isMyProfile={true}
    title="Achievements"
  />
  <ResumeEntry
    display="block"
    entries={
      Array [
        Object {
          "date": "2018-20",
          "subText": "Bachelor of Science - Major in computing and software systems",
          "text": "The University of Melbourne",
        },
        Object {
          "date": "2017",
          "subText": "Completed International Baccalaureate Diploma",
          "text": "School College",
        },
      ]
    }
    isMyProfile={true}
    title="Education"
  />
</div>
`;
